_posts/2017-02-02-Rust-1.15.md

```diff
@@ -0,0 +1,273 @@
+[37882]: https://github.com/rust-lang/rust/pull/37882
+[37855]: https://github.com/rust-lang/rust/pull/37855
+
+There were a number of functions stabilized as well:
+
+* [`std::iter::Iterator::min_by`] and [`std::iter::Iterator::max_by`]
+* [`std::os::*::fs::FileExt`]
+* [`std::sync::atomic::Atomic*::get_mut`] and [`std::sync::atomic::Atomic*::into_inner`]
+* [`std::vec::IntoIter::as_slice`] and [`std::vec::IntoIter::as_mut_slice`]
+* [`std::sync::mpsc::Receiver::try_iter`]
+* [`std::os::unix::process::CommandExt::before_exec`]
+* [`std::rc::Rc::strong_count`] and [`std::rc::Rc::weak_count`]
+* [`std::sync::Arc::strong_count`] and [`std::sync::Arc::weak_count`]
+* [`std::char::encode_utf8`] and [`std::char::encode_utf16`]
+* [`std::cell::Ref::clone`]
+* [`std::io::Take::into_inner`]
+
+[`std::iter::Iterator::min_by`]: https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.min_by
+[`std::iter::Iterator::max_by`]: https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.max_by
+[`std::os::*::fs::FileExt`]: https://doc.rust-lang.org/std/os/unix/fs/trait.FileExt.html
+[`std::sync::atomic::Atomic*::get_mut`]: https://doc.rust-lang.org/std/sync/atomic/struct.AtomicU8.html#method.get_mut
+[`std::sync::atomic::Atomic*::into_inner`]: https://doc.rust-lang.org/std/sync/atomic/struct.AtomicU8.html#method.into_inner
+[`std::vec::IntoIter::as_slice`]: https://doc.rust-lang.org/std/vec/struct.IntoIter.html#method.as_slice
+[`std::vec::IntoIter::as_mut_slice`]: https://doc.rust-lang.org/std/vec/struct.IntoIter.html#method.as_mut_slice
+[`std::sync::mpsc::Receiver::try_iter`]: https://doc.rust-lang.org/std/sync/mpsc/struct.Receiver.html#method.try_iter
+[`std::os::unix::process::CommandExt::before_exec`]: https://doc.rust-lang.org/std/os/unix/process/trait.CommandExt.html#tymethod.before_exec
+[`std::rc::Rc::strong_count`]: https://doc.rust-lang.org/std/rc/struct.Rc.html#method.strong_count
+[`std::rc::Rc::weak_count`]: https://doc.rust-lang.org/std/rc/struct.Rc.html#method.weak_count
+[`std::sync::Arc::strong_count`]: https://doc.rust-lang.org/std/sync/struct.Arc.html#method.strong_count
+[`std::sync::Arc::weak_count`]: https://doc.rust-lang.org/std/sync/struct.Arc.html#method.weak_count
+[`std::char::encode_utf8`]: https://doc.rust-lang.org/std/primitive.char.html#method.encode_utf8
+[`std::char::encode_utf16`]: https://doc.rust-lang.org/std/primitive.char.html#method.encode_utf16
+[`std::cell::Ref::clone`]: https://doc.rust-lang.org/std/cell/struct.Ref.html#method.clone
+[`std::io::Take::into_inner`]: https://doc.rust-lang.org/std/io/struct.Take.html#method.into_inner
+
+See the [detailed release notes][notes] for more.
+
+#### Cargo features
+
+Cargo has received an ergonomics improvement: If you have a file named
+`build.rs` in the root directory, it will [now assume that it's a build script
+][cargo/3361]. Previously, all build scripts required configuration, but this
+convention was strong within the community, so we've encoded it into Cargo.
```

Comment:
This isn't quite true just yet unfortunately. We'll eventually turn this on but right now it's a backcompat hazard. Specifically what's available in 1.15.0 is that Cargo will emit a warning if you have `build.rs` but *don't* have a `build = "build.rs"` annotation (asking you to write down `build = false` in `Cargo.toml`).

Eventually we'll turn on this auto inference, but we need a few cycles of the warning first